src/stepedge_nodes.cpp

```diff
@@ -2393,14 +2405,92 @@ void EptInPolygonsNode::process()
         std::cout << "Point (" << px << ", " <<py << ", "  << pz << ") is not in the polygon bbox.\n";
         continue;
       }
+      // For the ground points we only test if the point is within the grid
+      // index cell, but we do not do a pip for the footprint itself.
+      // The reasons for this:
+      //  - If we would do a pip for with the ground points, we would need to
+      //    have a separate list of buffered footprints that we use for the
+      //    ground pip. Still it would be often the case that there are no
+      //    ground points found for a buffered footprint. Thus we need a larger
+      //    area in which we can guarantee that we find at least a couple of
+      //    ground points.
+      //  - Because we work with Netherlands data, the ground relief is small.
+      //    Thus a single ground height value per grid cell is good enough for
+      //    representing the ground/floor elevation of the buildings in that
+      //    grid cell.
+      bool found_pip(false);
       for(size_t& poly_i : pindex_vals[lincoord]) {
-        if (GridTest(poly_grids[poly_i], point)) {
+        if (pclass == 2) {
+          point_clouds_ground[poly_i].push_back(pz);
+        } else if (not found_pip and GridTest(poly_grids[poly_i], point)) {
           point_clouds.get<PointCollection&>(poly_i).push_back({px, py, pz});
-          break;
+          found_pip = true;
         }
       }
     }
   }
+
+  // DEBUG: write polygon WKT with ground height to file
+  std::string fout("/tmp/geoflow_footprint_ground.tsv");
+  std::ofstream out_tsv(fout.c_str());
+  std::ostringstream wkt;
+  wkt << "geom" << "\t" << "ground_height" << std::endl;
+  // end DEBUG
+
+  // Compute median ground height per grid cell and store it for each polygon
+  std::cout <<"Computing the median ground elevation in the grid cells..." << std::endl;
+  for (auto& cell : pindex_vals) {
+    std::vector<float> z_vec;
+    for (size_t& poly_i : cell) {
+      for (float& pz : point_clouds_ground[poly_i]) {
+        z_vec.push_back(pz);
+      }
+    }
+    float median;
+    // Median for odd-number of elements
+    if (z_vec.size() % 2 > 0) {
+      size_t n = z_vec.size() / 2;
+      std::nth_element(z_vec.begin(), z_vec.begin()+n, z_vec.end());
+      median = z_vec[n];
+    }
+    // even-number of elements
+    else {
+      float m0, m1;
+      size_t n = z_vec.size() / 2;
+      std::nth_element(z_vec.begin(), z_vec.begin()+n, z_vec.end());
+      m0 = z_vec[n];
+      n++;
+      std::nth_element(z_vec.begin(), z_vec.begin()+n, z_vec.end());
+      m1 = z_vec[n];
+      median = (m0 + m1) / 2;
+    }
+    // Assign the median ground elevation to each polygon
+    for (size_t& poly_i : cell) {
+      ground_heights.get<vec1f>(poly_i).push_back(median);
```

Comment:
here a vector of floats is created for each building (with the median ground heights per overlapping grid cell?). We need only one float per building.